lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -363,22 +363,21 @@ export function usePaginatedAPIFetch<
   }, [fetchResult.data?.pagination.next, fetchResult.isLoading]);
 
   useEffect(() => {
-    // Every time data is loaded, append the result to the list
-    setCurrentList(prev => {
-      if (!fetchResult.data) {
-        return prev;
-      }
+    if (!fetchResult.data) {
+      return;
```

Comment:
All side effects depend on `fetchResult.data` being set, so we can simply skip the execution and keep previous state while it is not set.